### ISSUE TITLE
Colors for menus broken, cannot customize Maxstore Pro Theme at all

### ISSUE THREAD

**Issue by bpipal** [themeum/kirki#1405]
### Issue description:
Update taken today for 3.0.3, this completely breaks the ability to modify the Maxstore Pro theme and makes the default color of my menus white (since my background is white they disappear)

restoring plug-in to previous version resolves the issue.
### Version used:
(Did you try using the develop branch from github? There's a chance your issue has already been adressed there)
3.0.3

**Comment by rinkuyadav999** [comment]
Yes, If Typography field does not have color option, it adds color: #ffffff; Some other also reported it.

**Comment by aristath** [comment]
I'm still trying to figure out what happens here... 
See this post: https://wordpress.org/support/topic/your-plugin-messed-up-my-theme/#post-9253885

I'm going through all typography-related files and still don't see anything that could have caused this.
* In https://github.com/aristath/kirki/blob/3.0.3/controls/typography/class-kirki-control-typography.php#L310-L315 the color input is only added if `color` is defined in the control's `default` value.
* In https://github.com/aristath/kirki/blob/3.0.3/controls/typography/typography.js#L77-L85 the color value is only saved when the color input changes
* In https://github.com/aristath/kirki/blob/3.0.3/field/class-kirki-field-typography.php#L176-L178 the color value gets sanitized only if it already exists
* In https://github.com/aristath/kirki/blob/3.0.3/field/class-kirki-field-typography.php#L189-L206 there's no mention whatsoever of color
* In https://github.com/aristath/kirki/blob/3.0.3/modules/css/field/class-kirki-output-field-typography.php#L50-L52 where we output the CSS for the typography field, if there is no color in the value then it's skipped completely.

**Comment by rinkuyadav999** [comment]
@aristath I did not reported this issue because i can also not reproduce this issue. but some users told me about this issue.

**Comment by aristath** [comment]
> I did not reported this issue because i can also not reproduce this issue. but some users told me about this issue.

Apparently it only happens on update, not on new sites. One of the strangest issues I've seen...

**Comment by x-bull** [comment]
I have this issue on the localhost. After 3 hours of tests and dozen of installs, uninstalls, setup and plugin combinations no success to replicate it. I can confirm that this issue exists... :(

**Comment by rinkuyadav999** [comment]
@x-bull you can reset theme modes and then please test it. to reset mods: https://wordpress.org/plugins/customizer-reset-by-wpzoom/

**Comment by x-bull** [comment]
Yes this helps, this is not answer why it happened :-(. My theory is that the color was added there with some old plugin version and ignored till now.

**Comment by aristath** [comment]
> My theory is that the color was added there with some old plugin version and ignored till now.

That's what I suspect too. But I was unable to pinpoint where exactly that happened.
I tested the maxstore-pro theme but so far I was unable to replicate the issue.

I just pushed a possible fix for this: https://github.com/aristath/kirki/commit/030dc14ac0b16f572c57c007895acf249d43863d
That commit will check if a value is set in the defaults when the customizer opens, and if it's not, then it will unset it from the value.
That means that if anyone has this issue, they can open the customizer, make any change to the typography control and save. That _should_ take care of things.
